picard/ui/tagsfromfilenames.py

```diff
@@ -26,19 +26,72 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
 
 
+from collections import OrderedDict
 import os.path
 import re
 
 from PyQt5 import QtWidgets
 
 from picard import config
+from picard.script.parser import normalize_tagname
 from picard.util.tags import display_tag_name
 
 from picard.ui import PicardDialog
 from picard.ui.ui_tagsfromfilenames import Ui_TagsFromFileNamesDialog
 from picard.ui.util import StandardButton
 
 
+class TagMatchExpression:
+    def __init__(self, expression, replace_underscores=False):
+        self.replace_underscores = replace_underscores
+        self._tag_re = re.compile(r"(%\w+%)")
+        self.numeric_tags = ('tracknumber', 'totaltracks', 'discnumber', 'totaldiscs')
```

Comment:
Perhaps rename to class `_numeric_tags` as it isn't meant to be modified (as class attribute, one could subclass and modify it).